src/elements/Input/Input.js

```diff
@@ -88,7 +88,13 @@ function Input(props) {
   )
   const rest = getUnhandledProps(Input, props)
   const ElementType = getElementType(Input, props)
-  const inputProps = _.pick(props, htmlInputPropNames)
+  const inputProps = {
+    ..._.pick(props, htmlInputPropNames),
+    // React gives a warning if no onChange is given to a controlled input. We
+    // handle the onChange on the wrapping element, so we don't actually need
+    // onChange on the input itself but let's pass a noop to suppress the warning.
+    onChange: _.noop,
```

Comment:
That makes sense, although adding it to the `htmlInputPropNames` actually causes it to be called twice. I think we need to do a similar thing here as the `portalProps`, where any prop we pass through to the input isn't included in the `rest`. e.g.
```js
// Currently:
const rest = getUnhandledProps(Input, props)
const inputProps = _.pick(props, htmlInputPropNames)

// Change to:
const unhandled = getUnhandledProps(Input, this.props)

const htmlInputProps = _.pick(unhandled, htmlInputPropNames)
const rest = _.omit(unhandled, htmlInputPropNames)
```